src/app/_hooks/useSearch.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { NextServerSearchParams } from '@/types/searchParams'
+import { SearchableByTitle } from '@/types/searchTypes'
```

Comment:
Do we want to only to search by `title` - What about the `description` text, for example?

Reply:
For now, it's only by title, but we can change `SearchableByTitle` to the following if necessary.

```
export type SearchableByTitle = {
  title: string
  description: string
}
```

(And also rename it to match the changes)

Reply:
Yes! I think it would make sense to have `SearchableByTitle` for Events (and later on perhaps location?); `SearchableByTitleAndDescription` for Blog Posts; and `SearchableByTitleAndDescriptionAndTopic` for Ecosystem Projects. Not sure about those names - but you get the gist :-)

Reply:
Yes I see!

Reply:
Instead of using `SearchableByTitle` and the likes, I've added a `searchBy` prop to `useSearch` so that we can list the searchable fields there. And it's type-safe 😎

```ts
// Blog
const { searchQuery, searchResults } = useSearch({
  searchParams,
  entries: posts,
  searchBy: ['title', 'description'],
})

// Events
const { searchQuery, searchResults } = useSearch({
  searchParams,
  entries: events,
  searchBy: ['title', 'location'],
})
```